packages/admin/components/forms/CollectionTypeahead/CollectionTypeahead.tsx

```diff
@@ -1,77 +1,105 @@
-  { data, control, name, label, disabled, required }: Props<T>,
+  { control, name, label, disabled, required }: Props<T>,
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   ref: Ref<HTMLInputElement>
 ) => {
-  const optionsData = useMaybeFragment(fragment, data);
+  const [q, setQ] = useState("a");
 
-  const options = useMemo(() => {
-    return optionsData?.viewer.collections.nodes.map((node: CollectionNode) => {
+  const formatOptions = (data: Response) => {
+    if (!data || !data.search?.results?.edges?.length) return;
+
+    const results = data.search.results.edges;
+    const options = results.map(({ node }) => {
       return {
         label: node.title || "",
-        value: node.id,
+        value: node.entity.id ?? "",
       };
     });
-  }, [optionsData]);
+    return options;
+  };
 
   const { t } = useTranslation();
 
-  return options ? (
-    <Controller<T>
-      name={name}
-      control={control}
-      rules={{
-        validate: (value) => {
-          return !!value || (t("forms.validation.collection") as string);
-        },
+  const debouncedOnChange = debounce((value) => setQ(value), 500);
+
+  return (
+    <QueryWrapper<Query> query={query} initialVariables={{ query: q }}>
+      {({ data }) => {
+        return (
+          <Controller<T>
+            name={name}
+            control={control}
+            rules={{
+              validate: (value) => {
+                return !!value || (t("forms.validation.collection") as string);
+              },
+            }}
+            render={({ field }) => (
+              <BaseTypeahead
+                label={label}
+                options={data ? formatOptions(data) : []}
+                onInputChange={debouncedOnChange}
+                disabled={disabled}
+                required={required}
+                defaultValue={q}
+                {...field}
+              />
+            )}
+          />
+        );
       }}
-      render={({ field }) => (
-        <Typeahead
-          label={label}
-          options={options}
-          disabled={disabled}
-          required={required}
-          {...field}
-        />
-      )}
-    />
-  ) : null;
+    </QueryWrapper>
+  );
 };
 
 interface Props<T> extends Omit<TypeaheadProps, "options" | "name"> {
-  data?: CollectionTypeaheadFragment$key | null;
   control: Control<T>;
   name: Path<T>;
 }
 
-type CollectionNode =
-  CollectionTypeaheadFragment$data["viewer"]["collections"]["nodes"][number];
-
 export default CollectionTypeahead;
 
-// Currently limited to 50 collections per query
-const fragment = graphql`
-  fragment CollectionTypeaheadFragment on Query {
-    viewer {
-      collections {
-        nodes {
-          id
-          title
-          slug
+const query = graphql`
+  query CollectionTypeaheadQuery($query: String!) {
+    search(visibility: ALL) {
+      results(
+        prefix: $query
+        page: 1
+        perPage: 50
+        order: TITLE_ASCENDING
+        schema: [
```

Comment:
Do the schemas need to be hardcoded here?

Reply:
Right — yes, that was one of my questions too. We can certainly pass them in as an argument, but how do we determine, for a given instance, what the full set of collection/item/etc. schema's is? That would be a prior query, right?

Reply:
I think I see... Is the issue how to search for just collections? I had to look at the Collections page to remember! You'll need to add the scope filter, `scope: COLLECTION` in this case. If we do need to filter down further by schemas, then we would need to make a prior query on the parent.

Reply:
I'm not sure that works because there isn't necessarily a parent, e.g. [here](https://wdp-admin-git-ld-wdp-750-cast-iron.vercel.app/users/GD7rv85CxX97UBONt8BoeSNbEdL2Xk6/collections?order=RECENT&drawer=addUserCollectionAccess&drawerSlug=GD7rv85CxX97UBONt8BoeSNbEdL2Xk6). And we don't want to limit it to direct descendants like `scope` does.

Reply:
Where are you looking at the Collections page? I think I'm missing something.

Reply:
I looked at `CollectionListSearchFragment` - `scope` isn't limited to direct descendants, unless you add `maxDepth: 1` to the search filters!

Reply:
(See `packages/admin/pages/collections/index.tsx`)

Reply:
Oh! The debug site confused me. It says: "This will include only directly descending Collections, no links."

Reply:
I will make this change then.